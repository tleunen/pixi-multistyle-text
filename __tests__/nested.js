it("should render a simple example correctly", () => {
	require("./support/pixi-shim");

	const MultiStyleText = require("../dist/pixi-multistyle-text");

	PIXI.settings.RESOLUTION = 2;
	let renderer = PIXI.autoDetectRenderer(400, 80);
	renderer.backgroundColor = 0x333333;
	document.getElementById("root").appendChild(renderer.view);
	let stage = new PIXI.Container();

	let nested = new MultiStyleText("You can <outline>nest <b>tags <red>as <i>deeply <thicker>as <shadow>you'd <large>like</large></shadow></thicker></i></red></b></outline>",
	{
		"default": {
			fontFamily: "Arial",
			fontSize: "24px",
			fill: "#cccccc",
			valign: "bottom"
		},
		"outline": { stroke: "black", strokeThickness: 2 },
		"b": { fontWeight: 700 },
		"red": { fill: "#ff8888" },
		"i": { fontStyle: "italic" },
		"thicker": { strokeThickness: 6 },
		"shadow": { dropShadow: true, dropShadowColor: "#888888" },
		"large": { fontSize: "36px" }
	});

	nested.x = 200 - nested.width / 2;
	nested.y = 20;
	stage.addChild(nested);

	renderer.render(stage);

	let canvas = document.getElementsByTagName("canvas")[0];
	expect(canvas.toDataURL()).toMatchSnapshot();
});