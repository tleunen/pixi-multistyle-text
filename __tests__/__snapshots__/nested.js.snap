exports[`test should render a simple example correctly 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAyAAAACgCAYAAAD0KAgtAAAABmJLR0QA/wD/AP+gvaeTAAAgAElEQVR4nO3deWBTZb438G+2pk33tE2hi7TsLRTFgkAFKZVt2ClVBxgvKIyKiM7oVbky96KDiuK4vCo4bDqyC7IILi8IKrtAsbKV0mUK3femSZM0+/sHyblJmxbUcpx5/X7+guTknJOcBJ7veX7P80hSU1OdICIiIiIiEoH01z4BIiIiIiL67WAAISIiIiIi0TCAEBERERGRaBhAiIiIiIhINAwgREREREQkGgYQIiIiIiISDQMIERERERGJhgGEiIiIiIhEwwBCRERERESiYQAhIiIiIiLRMIAQEREREZFoGECIiIiIiEg0DCBERERERCQaBhAiIiIiIhINAwgREREREYmGAYSIiIiIiETDAEJERERERKJhACEiIiIiItEwgBARERERkWgYQIiIiIiISDQMIEREREREJBoGECIiIiIiEg0DCBERERERiYYBhIiIiIiIRMMAQkREREREomEAISIiIiIi0TCAEBERERGRaBhAiIiIiIhINAwgREREREQkGgYQIiIiIiISDQMIERERERGJhgGEiIiIiIhEwwBCRERERESiYQAhIiIiIiLRMIAQEREREZFoGECIiIiIiEg0DCBERERERCQaBhAiIiIiIhINAwgREREREYmGAYSIiIiIiETDAEJERERERKJhACEiIiIiItEwgBARERERkWgYQIiIiIiISDQMIEREREREJBoGECIiIiIiEg0DCBERERERiYYBhIiIiIiIRMMAQkREREREomEAISIiIiIi0TCAEBERERGRaBhAiIiIiIhINPJf+wSIiIjo35vdbkdtbS20Wi3MZjOcTicAQKFQIDg4GBqNBv7+/gCAxsZG1NXVwWAwwG63AwBkMhlUKhUiIiIQEREh7NdgMKC2thbNzc2w2WwAAKlUCplMBpvN5nUch8PRZn+RkZEIDQ0V7XMgopsjSU1Ndf7aJ0FERET/noxGI4qKimCxWNrdRiKRIDY2Fs3NzdBqtR3uLzg4GD169EBVVRWqqqp+8fmp1WokJCRAIpH84n0RUedgACEiIqKfxWq1Ijc3V+id6Cz+/v5oaWnptP1FREQgISGh0/ZHRL8MS7CIiIhEYLfbYTabYbPZIJfL4efnB7nc93/DFosFZrMZDocDcrkcSqWy3W1bHwO43uMglXoP83Q4HELJkkwmEx43mUywWq2w2+2QyWTw8/MTyqVutN/y8vJODx8AOjV8AEB9fT0iIiKgUqm83rvVaoXVaoXNZoNCoYCfn5/X80R0azCAEBER3SJmsxm1tbWor6/32VCXy+WIiIhAVFQUrFYrKioqYDAY4HA4vLaTSCRQqVRQq9WIioryKieqq6sTxlR4UigUCA8PB3B93IXVavXaX0BAgNAAb02hUCAkJAQWiwXNzc1CcPF8zmAwdHpQuJXy8/MBXB9DEhgYCJvNBpPJ1Ga7gIAAREREQK1WQ6FQiH2aRL8JLMEiIiLqZHa7HSUlJWhsbPRqvHcGf39/3HbbbQgMDERhYSH0en2n7p+uk8lkiI2NRVRU1K99KkT/3+E0vERERJ3IYrHgypUraGho6PTwAVwvTyooKEB+fv6/ZPgYGBODhqVL8ZeMjBtum/Pkkyh+/vmb2u+HWVloWLoUmqAgAMDI7t3RsHQp/vvee4Vtzi5ahGMLFvy8E2/FHSKLiopuyXUk+i1jACEiIuokTqcTRUVFPkt7Ovs4rUuu/lUYrVaEBwTA4ho30lkC/fwQHhAAqav8TCGVIjwgAP4eY2NC/f0RolR26nG1Wi1KSko6dZ9Ev3UcA0JERKIxGAwoLi6G2WwGcL3ePjExEQEBAb/qedXX16OiogJ2ux0SiQTBwcFISEhoM5D7Rqqrq2E0Gm/RWf57aHKNC9Hd4vEhh4uLoX7pJbTcgkHwrdXV1SEwMBCRkZG3/FhEvwUMIEREuN4ALS8vh9VqhUQiQVBQEBITE9sdhFpaWoq6ujo4HA5IpVKo1WrcdtttXGvgBsrKyoTwAVyfgam8vBw9e/b81c7JZrPh2rVrXmU2jY2NUKlU6NKly03vx+l0orq6ut3n/eVy9IuORlRgIGoMBuRWV7fbeI4NCUGSRgN/hQJ5NTUorK+/qXMIViohl0qhM5thbzWQXSaVIkSphM3hgN7jGvjL5RjWrRsiVSrUGgz4vqSkzXkF+vnBTybzud+wgADYPfbpDh46j2MAgEImQ7JGA4VMhryaGjR3sG7IzbDa7Wi8yZ6mID8/KGQytNhsMLUajJ/SpQs0gYGo0OmQW1PT7j4qKysRERHB3zhRJ2AAIaLfPIfDgZKSEmHmIafTCb1ej+rqasTFxbXZXq/Xo8ajoeJwOFBXV4fQ0FCEhYWJdt7/jtyL1RU//zy0JhMGvvuuz1mYxGS1WuF0OpGk0eD4ggVYf+YMnv3yyw4X1vNFr9e3OyXt0yNGYHF6OqICA4XHGk0mvHX0KF7+5hvhMT+ZDKumTcNDgwYJpUYAsD8/HzO3br1hg/uNCRPw6JAhePjTT/FRdrbXc7PuuAMb7r8f/+f4cfxp3z4AwAMDBuD9qVMR6XFetQYD/rRvH7b8+KPw2PtTp2JuaipS3n4bF1uFrIoXXsDF6mrc9f77AK6XYFntdq8Acl9KClZOmya8f73ZjCf37u3wvdzI6J498fX8+Xj122+xZP/+drcb06sX9vzHf6C2uRmj1q5FcUMDgOtjSNZmZqKXR6/GpepqLNizB0eLi9vsx2KxoKGhwWuldiL6eTgGhIj+ZdjtdmG9AYfDAYPBAJ1OB5PJ1OEgUJPJBL1eD4PBILzeF6vVCr1eD61WC4PBIOzTZrPB4XAgOjoa2dnZWL58OQC02wB1N5gnTJiA7Oxs3H///V6Pu1ksFuj1euj1epjN5g7fg8VigcFguOF2TqcTLS0t0Gq10Ol0P7mRDFz/nI1GI3Q6HYxGY4efmfs9NDU1/ezSIvfxbvb1N3s9gesDsvV6PZqbm2+4HoXD4YDJZBLed+vPWSaRIDwgACpXr5fT6YTdbvd5PdzPeZ5fc3Ozz+NO6NMHb06ciAqdDtM2bMCg997DA1u2oFKvx7KxYzE5KUnY9qUxYzBv8GC8ceQIhq1ahYy1a/HeiRMY17s3Vk6b1uH7A4BNOTkAgKyUlDbP3ed6zL3NxL59sWXmTNQbjbh/82YMeu89zNq6FVqTCZseeEDY/uc4dvUqKnQ6AEB69+7YOnMmVAoFXjp4EDO3bsXHZ89i3YwZ6HOLZ5ga37s39s6Zgyq9HiPXrBHCx5D4eOx/+GGoVSo8/fnnGLd+PZ7atw+aoCAcmDcPg33ceAAAnes9EdEvwx4Qok7St29fPPLIIzAYDHjllVfanR9/0KBBmDVrFurr67FixYpf/e7vvwKz2YyCggKhNMe9EJhn406hUKBLly7QaDQA/rfcpaamps36BqGhoYiNjRUWU2tubkZZWVmbQbsymQxRUVHtNhwbGxtx9epVrxWU3SVDvpSVlQmN1ZqaGq9SI/d7iIyMRHR0tPAetVotKioqvAYty+VySCQSOBwOSCQShIWFIS4uDjU1Naiurm7TKFepVIiLi0NwcLDP83IzGo2oqqqCVqv1alS7P7Po6GgEuWYY0mq1KC8vb/M9lsvlwnW4USlKS0sLysrKoNPpbjiL0E+5nsD1mvyqqiqvz9g9diMmJgYBAQEoLi5GU1MTnE4npFIppFKpV0iRy+WIjo5GSEiIz3Nyr68hkUgQGxuL6OhoANevc01NjfCelEolevXq1e5veXJyMgAga9MmoZTqbHk56o1GHJw/H3fFx2Pf5csAgGn9+qGmuRmLv/pKeP23RUXoFhaG9O7dIZVI4Ojgszx+7RqKGxowumdPhAUEQOv6XoX6+2Nsr17Iq61FdlkZpBIJ3pk8GXaHA+M+/BDXGhuF8zpRUoK8Z57BGxMmYOfFix0erz0Za9cKf35z4kRIJRKMW78ex69dAwBsO3cOl6qr8cH06T953zdrYt++2PmHP6BEq0XG2rUoa2oSnvtg+nRIJBKMWrMGF6qqAAAHCgpwqqQE3y9ciJfGjMGEjz5qs8/2/q0gop+GAYSok+Tl5SE/Px8DBw7ExIkTsXPnzjbbKJVKZGVlITAwEP/4xz8YPlwqKythNpuFVYr1ej0kEgl69+6NsLAwVFZWorS0FKWlpbBYLOjatSsKCwuFxkDXrl0RHx8Pg8GAK1euCL0DCQkJcDgcuHr1KgAgPDwciYmJCAoKQnl5OYqKilDlanwolUqh8e1eaM1kMqG+vh6RkZHCcxUVFbBYLFCpVEKDWKlUIiQkBHq9HqWlpcL70mg0uO222+B0OlFZWYmKigpUVlaisbERffv2RVVVlXD80NBQxMfHo7GxsU3Aqaurg1arhc1mg0QiQffu3REXF4eWlhbk5eVBp9OhoKAACQkJUKvVPj/jxsZGFBcXw+l0+hyLoNVq0dTUhG7dusFisaCiogKA77EIZWVlMBqNSExMbPea6vV6FBYWwuFwQCmXI9kVWPJqamBs9b232+3Iz88Xrme38HD0jIiA3mxGTkWF1/UMCwtDcXExGl0N5q7Bweir0cBssyGnogI6nQ56vR5BQUFeU9Q6HA44HA70j45GXGgoSrRa5NbUoLy8HA2uu+KthSiVkEmlaGppQXl5OdRqNaxWK6qrqyGXShHs7w+r3Y5msxlVVVXthqylX3+NNw4fxj9bHScuNBQAIPcY6C6VSKAJCsKfhg/HmlOnhM9q6oYN7X7WnpxOJ7b8+COWZGRgSlISNvzwAwBgclISlHI5Nrt6P+6IiUHPiAjsunhRCB9u1xobcbCgAJOSkpCs0bQpufopekdG4s7YWBwsLBTCh9v67GysmDDhZ++7I1OSk7F91iyUNjUhfc0aoTcGAPpFR2NgTAz2XLokhA+3U6WluFBVhVE9ekAikbS5prdi1Xei3yIGEKJOtGXLFvTp0wcZGRk4ceIEKisrvZ6fNGkSwsPDcerUKZw/f/5XOst/Pe7/1JcvX47U1FTMmzcPy5YtQ/fu3YVtjh07hueeew7V1dXQ6/UwGo3o2rUr/ud//geDBw8WtmtoaMAHH3yA3bt3C8FDIpHg8ccfx+zZs+Hn5ydsm5ubi+eeew5VVVWYP38+HnroIQBAeno60tPTsXjxYhw8eNArKLrP9f3338eAAQMAAE899RSeeuopjB07Fg0NDdBoNFiyZAnS0tK8egkuXbqEV155Bfn5+bhy5QpMJhPkcjmefvppzJgxQ+gVOX/+PFavXo3ly5fjgw8+wI4dO2Cz2dCtWzcsW7YMya476sD1sq/169dj3bp1KCkpQUhICORy73/aPQdZ32gsgnsszI3GIjQ0NCA0NNRn4LFarSgqKoLD4cBDgwbhzYkTEe6a5UprMuGPu3Z5bW82m2E2m9EtPBwfZmUho0cP4bma5mb85cABrD19GlevXkVwcDCampqgVqmwNjMT0/v1Ez7jppYWvHHkCJZ/+60QPpZkZODlsWNx3+bNWJSWhns8QtOhwkLM2rYNNe3c1V4zYwYeGDAAmRs3YvelS16riS8YOhTvTpmClw4exIsHD/psrHq+B/+wMDx1991IjY1FolqNhPBwIYB4+vO+fdj14IN4e9IkLBs7FseuXsXXBQXYm5t70wPRN+XkYElGBrJSUoQAkpWSIoQTAEh29SSe9Qi7fn5+Qkmf+1jxYWG/KIAM6NoVANo09IHrg8ivNjYiuJOnzU0ID8eO2bPhJ5Mh1N+/zXTAA1wTCyRHR2P77NltXh+hUsFfLkeIUinM6OXmDrI/dXY0IvLGAELUiXQ6Hfbs2YNZs2bhgQcewDvvvCM8FxMTg3vvvRdGoxGffvqpz9fHxcUhKioKdrsdV69ebVNvLJPJ4O/vD4vF0qb3RKlUQi6Xw2QyCYOpOyKXy5GYmIjg4GBhnntfd/fc5SdhYWFobm4Weita70upVArHjoiIQFRUFFpaWtrMLnQjCoUCH3zwAc6ePYuPPvoISqUSs2bNwvDhw5GVlYUtW7bAaDQiJCQEa9asgUajwcaNG5GdnY2QkBA8+OCDWLJkCeRyOXbs2AEAyMrKwkMPPYS9e/fis88+g1QqxZAhQzB37ly8+uqrePjhh/Hdd9+hubkZixYtwuXLl7F7925cdpXFeDY23I3djz/+GCNHjsSUKVOwf/9+nD17VijxevHFFzFo0CBs3LgRR48ehd1ux7BhwzBv3jy89tpryMzMFEqulixZgsmTJ+P06dP44osvEBAQgMzMTLz22msIDg4WApNUKsXf/vY3REREYOnSpfjnP/8JjUaDOXPm4LHHHkNxcTEOHTqE2tpadHU1+twaGxtht9uFsQjnKivxx507UdbUhB4REVg6ejSWjR2Lc5WVQimQeyzC64cPY8+lSwhQKDC9Xz8sSkvDymnTMGvrVlRXV/sMIO7pbKckJ2P9jBloamnBXw4cQFF9Pe7t2RNbZ86E3eEQyoMAQK1S4fAjjyA2NBR/O3IE3xYVIVylwn+OGIE1mZlQyGRYdfIkmpqaoJTLcWDePNwZE4N1Z87g87w8+MlkeGLYMLw8dixClEo871HCBABrMzORU1GB6Rs3wmyz4cE778TM22/H53PnIm3VKth8/GY25eTggQEDkJWSgt2XLkGr1QoNdPf4iM2uBn1YWFi7pXlZKSn4+P77oVIoUFBXh3OVlfimqAhVej1WtRrX8eWVK+j75puYk5qKiUlJGN2zJ8b37o03JkzA+jNn8Oju3Tf8PeXV1uJseTnG9uolrIkxrndvHL92TeiFCXR9r7SuBrZCoYBGo0FlZSXsdrtX6OyIRCKBnys4++I+TuuGvJvRau30ABKsVOKH8nIcKirCs/fcg3cmTcIfPvmkzTm5x/y0lltTg9yaGijlvptINpvN60YGEf10DCBEnezo0aMYMmQIkpKSkJqairNnzwIAHnjgAchkMnz66adtgkVUVBTmz5/vNdbA6XTiyJEj2L59uxAMkpKSsGjRInz++efY55rFxm3mzJkYNmwYXnjhBdTf4E5pUlIS5syZg/DwcOExrVaLbdu2IcdVogEAgwcPxvTp071mfbFYLDhx4gS2b98ujEVIT0/Hfffdh5UrV2LEiBFCzwBwfV2ElStXdjg9qSeZTIaTJ09iyZIlwmN5eXnYtGkThgwZgi1btgAAHn30UXTt2hUvv/wy9uzZI2x77Ngx7Ny5EwsWLMCePXtgtVoxcuRIOJ1OvP7660J4ysnJQUhICMaPHw+1Wo1Lly6hrq4OixYtQllZGXa57tL7+fkJ5VcAEBERAb1ej8OHDyMwMBBTpkzBuXPnhO0DAgIwePBgHD16FO+++67wuvPnzyM5ORl33303QkJCoNPpkJycjMmTJ+Ps2bN44oknhOC4d+9ebN682WtMR0JCAhITE7Fz50588cUXAIDLly8jJycHu3btQlpaGg4dOuRz7JF78Hdnj0XwdSyHw4GGhgZIJBK8OXEi7E4n7lm9WrgDvu3cORTV12P5+PFer3tx9Gh0Cw/HH3fuxLozZ4THv8jLQ94zz2DZmDFYd/o0LHY7nhg2DKmxsXj5m2/w3wcOCNvuzc3FD08+iT8PH463jx1DlUcJVqVej/EffijcDf/qyhWoFApMTU7G5KQk7L50qc172Z+fj1qDAZOSkuAvlwu9Kl2Dg3F3QgJOlZaioK4OCoUCwcHBPj8PP5kM67OyYLHZMHL1amSXlQnP3dNOCdvVxka8dPAgXjp4EGEBAZiclISl996LP951Fz7Mzsb3N7Eo3qacHLw9aRImJycLZXcbXb0hAITZtGJc3zGr1Yoyj3Pr7gqWJVpth8fRBAZC1kFvQIPru+ert8f9uLWTFyws1+kwas0amGw2/K5PH8weOBBbfvwRX165AgCod53T/vx8LPzss5+8f/Z+EP1y/BURdTKHw4GNGzfCZrPhvvvug1KpxKBBg9C3b19cuXIFJ06c8No+MDAQTz/9NOLi4rB7924sX74c77zzDnJycjBy5EjMmzevU8/vtttuw8KFC6FUKrF582a8+eab2Lp1KxQKBebPny8stBUTE4N58+ZBIpFg/fr1ePXVV7Fy5Upcu3YN6enpGDVqVJt9P/jgg1Cr1di8eTPWr1+P8+fPIzo6Wpgl6mZtcNW7x8TEQCKRID8/Hw6HQwhMEokE48ePh1arxWetGhB6vR6HDh1CSEgIevXqBeB6g0EikeCRRx7xatSvWLECGRkZ7Y4BCAoKQnJyslAaBVwPIL6m5nUzm82YNm0aXnrpJa/HFQqFcP7uEqn09HQAwPbt2+FwOBAVFQW1Wg2LxYJt27Z5vd7d8zJy5EikpqYKf9fpdBg9ejSWLVsmvNfW3Nsu/fpr9Fixok0pz43GIqg81kKZumEDYl55BQ7XwO7WDAYDHA4HBsfFoWdEhFBn7+/vj27dugEA3jtxwmstCYlEgll33IE6gwHrW00dqzWZsPPCBahVKtzu6tn5w8CBcDideOPwYa9tLXY7NvzwAxQyGYZ7hHkA+Cg7Gxa7HWq1Wugh+vv33wO4Pp2rL1a7HdvPn0eIUomxvXsLj89ISYFUIhFmk1Kr1TCbzT57HhPVaoQolThSXOwVPgAgNTbW6+/dwsNR9Nxz+C+P35bWZMLGH37ACtd7TfS4adCRbefOwe5wIKt/f9yXkoIWmw3bL1wQnv++pAROpxNTPUrY3IKVSmT07IlKvR55tbUAAKOr50etUnltO75Pnw7P43RpKewOByb27dump2RIfDxi25kA4JdoMBqhM5thtduxwNVj9PfMTKGn5VRpKRxOJzJ69mzT06OQyfD9woXY386/u1KptE2JIxH9dPwVEd0ClZWV2L9/PyZOnIjMzEzccccdsNls2Lx5c5vyiTFjxkCtVmP79u04dOiQ8HheXh7+/Oc/484770SPHj1QVFTUKeeWmZkJhUKBVatWITc3FwCQn5+PlpYWPPTQQ7j99ttx6NAh9HM1TDZv3oyLFy8Kry8vL8err77qcwCy1WrFihUrhF6Gs2fP4vXXX0efPn06rJFvzb3GRlRUFCorK+FwOGCz2YT/+KOiohAaGgqdTidMmevJ3dB1lwetXLkSycnJmDNnDmbOnIkLFy7g1KlTOHr0KAoKCto9D4VC4RU+PB9vj8PhQEVFBYYOHYrBgwcjISEBXbt2RWxsLFStGm/uHi/3OWg0GthsNjQ0NAjjV9yKioqwfft23H///Vi9ejVqamqQnZ2NkydP4ujRo8IAbl8lUeHh4aitre30sQjhPhrD7mvvrrN3jx9Qq9WIjIxEZWUlDBYLKjx6J2JDQhChUqHRZMIns2a12WcfVyjWBAVBIpGgf5cusNrtWJeV1WbbeNd70Xj0WgHAeVcPjFqtRkBAACorK4VF59q7Ow9c70lYOGwYsvr3x17X7yWrf39Y7XZ8cu6csE8/Pz+f3/HypibYHQ6kxsUhOigI1c3NkEgkmJqcjJfGjAEAoWFeqtXCXy7HwmHDsOmHH1DqmrVJKZdjar9+cDidXmM2OlKl1+NgYSHGuYLT55cve5W8lWi12Hf5MqYkJ+PZe+7BG0eOwOl0Qi6V4q1Jk6BSKPDXgweFoHjZFUSevPtunCkrg8lqxV3x8Vg+fnyHs2RVNzdj27lzmD1wIF4dPx7Pf/UV7A4HugQH39IZsNyOXb2KD7OzMW/wYCwfPx5PfPYZKnQ6bDt3DrPuuAOL09Ox/Lvv4HQ6oVIo8MH06RgSHy+sk9JaUKvvFRH9PAwgRLfIl19+iUGDBgl3uffs2eOzDOnOO++ExWLB8ePHvR53Op04fvw4+vTpg379+nVKAFGpVOjTpw/q6+uF8OH2448/YteuXcKsTMePH0dOTk6b3gH3Qnu+7n4fPnzYa3yI3W5HeXk5+vbtC39/f6+pZjvi3kd7pQ4BHnXbvqZQbWxsxOnTp4X95ObmYsaMGZg0aRKGDx+OlJQUpKam4vHHH8eRI0ewePHin7Wehi9yuRxvvfUW0tLSYLFYcPHiRVy8eBGffPIJMjIykJaWJmyrdN2RdZfuyGQyoazNVznPihUrcOjQIYwbNw5DhgzBhAkTMGHCBBgMBrz88sv4+uuvUVtb22Y6Xvf37ueORfhdnz4+xyLU19cjPj7e6w66uxdAGGPguubuIOe+pkaPz9u9rdPp9FmTX2Mw4GBhIUxWK5QyGeRSKSx2u89tmy0WHCwsRGOrNUf0raZ49tRRA/pUaSkK6+sxJTkZfjIZ1CoVhick4KsrV1BrMCAgIEAIlkqlss11a7ZYsOLwYfzXqFEoePZZXKyuRrewMKhVKvxp3z78beJELBg6FDKpFH/atw9P7t2L7bNn48p//icuVVej0WTCwJgYRAYG4rXvvkN+XV2759rappwcIYCs9yhrc3ts9270jorC67/7HRYOG4arjY3oExWF6KAg7Lt8GW8dO+a1r6eHD8eM/v0xJSkJjSYTNEFBWHP6NEZ5TBbhy5P79qF/ly54ZsQIPDhwICr1eiRpNML3r3s7M7d1lue/+gpTkpOxYOhQbDt3DseuXsXCzz5Dz4gIvDJuHJ4eMQLXGhvROyoKQX5+2HruHN5t1VPt5it0E9FPxwBCdIvYbDbs2LEDTzzxBOrq6nDAo1bdTS6XQ6PR+FxvAQBqXXcd25ta9afq0qULpFIpSnzUkLe0tGC/x2rCRqMRUqkUw4cPR2JiIiIiIhAZGdnhf8C+Spncg+V9Nfx+ribXneGamho8/vjjN/WahoYGbNiwARs2bIC/vz/S0tLw8MMP45577kF6errP6/NzTJs2DWlpaThw4AD++te/el3X4cOHe23rHlOgVquFdUPcA9nj4+N97v/s2bPCuKL4+HiMGzcO8+bNwzPPPIOvv/66zToFTqcTOp3ulo1FMBqNCPSYUct9nRtcwcPdu+CeoMBisUAikSAuNBQFrsa0uya/XKfDmHXr2v1s3QwWC+xO50qoiQ8AAAgGSURBVE1t6+YuHbJarULIc/eWVHewtoPT6cTmnBwsHT0aY3r1QrfwcMikUmz0KL9yU6lUPn/HL+zfj3OVlZjarx+C/fzwXVER1mdno6i+HkX19cjs3x8/uHo2dl68iJGrV2Nuaip6RUZCLpXis9xcfJGX53OcSkc+y82F3eFAiVaLAz56+ir1egx67z08OmQIRnXvjhB/fxwsKMDey5ex48IFr94crcmEwe+/j0eGDEE/jQZNLS34Ii8PX165gt/ffjvMHUxP22A0Im3VKjyRloZRPXrAZrdjU04OVp48iZGJiT6DpC9rT5/GocJC6Fyf8eWaGjy6axdyXFNGA8BfDhxoM6FAvdGIqR9/jJQuXYRyQq3JhOF//zseSk3F+D59EOTnhx3nz2P3pUvCOKjW3GvyENEvxwBCdAu5B4PrdDqfKzorlUpIJJIb9gzcTOnSzdQlu++438yK1D169MDChQsRGBiIxsZGlJSUIDs7G8XFxXjsscd8vkasdU3cs3YlJCRAo9EIJVtuL7zwAgYPHoy5c+fCZDJh27Zt+P7777FixQoA18PWN998A5PJhPfeew8xMTE+j9PU1ISysrIOx3y01tt1x3nnzp1ejVGpVCo855aXl4fx48cjLS0NeXl5KCgoEHoQxo0b57Xt9OnTMXfuXDz//PPIy8sDAJSWlmLdunW4/fbbMXToUGEaVc9pQt0rq7vHIuzNzb2psQjf/PGPWHfmDJZ/+y2A/x2LECCXY3VmJhLDw/F9SQnMZrNXAHH3Tp10rfkwNTkZz3/1Ferr66HX6+FwODCud2+h1wMA6gwG5NfVoW9UFOJCQ70WjAOAv0+fjnt79sTQlStRbzTixLVrGNOrF+6Kj8dpj3VXAOC/770Xc1NTMeXjj3HJo8dxTM+e+PzyZa/wPcM1k9WJVutTuPn7+6OlpQWbf/wRS0ePRlZKCrqFhaGppUUox/IMIBEREe2OJ/rk/Hl84mPq7YOFhThYWOj12LGrV3GsVQnezzEoLg4yqRRrz5xpt5fHYLHgraNH8dbRozfcX73RKHwfPG1zlaJ1xGi1YsXhw8JYFrf/m59/w9e6tf6cynU6rDl9+qbO5WRJCU62uvFitdux5vTpNvtoT0pKCsd/EHUSDkIn+hWZTCbY7XaEtlOD7l71u7HVQmG+VqBubx+e3HfHfW2rUCiQmpoqjEuYPXs2VCoV3n//fSxevBirVq3Crl27UFxcfMPjiGHz5s2Qy+V49tlnhQUBJRIJfv/732P69OkoKChAU1MTLBYLdDodJk2ahB4e60tIpVJkZGQAgFCO5g5QGo0GMpkMDocD1dXVbVZQd3/+7hKvLq7xDgCEEjbPUquQkBC8+OKLQtBxN2K+/PJLGAwGzJkzByNGjABwvQG/YMEC4e9uRUVFiI2NxZw5c7zGoMTFxaFv374oKiqCxWKBv7+/V+mae7rQ1mMR3O9jWr9+HY5FiPf4rvgaixDQ6u51QEAA/Pz8UFhfjy/y8tArMhJ/yciAxPV5dQsPxzuTJ6O1t48ehUImw7tTpgh3qSUSCZ68+248ctddOFdZKfSUvOMqDXp70iSvu+f3DxiAJaNGod5oFMZ3uD0yZAjG9uoFm80Gm82GKcnJeGzIEFQ3N2OnxxgnT8HBwVC4ytVOlZYis39/3JOYiB0XLqDFZvOaJhm4fp3dkzj8mqQSCaQSCRanp6PZYsHam2xgU/siIiIwY8YMn//2EtFPJ0lNTb35CfqJiIiIiIh+AfaAEBERERGRaBhAiIiIiIhINAwgREREREQkGgYQIiIiIiISDQMIERERERGJhgGEiIiIiIhEwwBCRERERESiYQAhIiIiIiLRMIAQEREREZFoGECIiIiIiEg0DCBERERERCQaBhAiIiIiIhINAwgREREREYmGAYSIiIiIiETDAEJERERERKJhACEiIiIiItEwgBARERERkWgYQIiIiIiISDQMIEREREREJBoGECIiIiIiEg0DCBERERERiYYBhIiIiIiIRMMAQkREREREomEAISIiIiIi0TCAEBERERGRaBhAiIiIiIhINAwgREREREQkGgYQIiIiIiISDQMIERERERGJhgGEiIiIiIhEwwBCRERERESiYQAhIiIiIiLRMIAQEREREZFoGECIiIiIiEg0DCBERERERCQaBhAiIiIiIhINAwgREREREYmGAYSIiIiIiETDAEJERERERKJhACEiIiIiItEwgBARERERkWgYQIiIiIiISDQMIEREREREJBoGECIiIiIiEg0DCBERERERiYYBhIiIiIiIRMMAQkREREREomEAISIiIiIi0TCAEBERERGRaBhAiIiIiIhINAwgREREREQkGgYQIiIiIiISDQMIERERERGJhgGEiIiIiIhEwwBCRERERESiYQAhIiIiIiLRMIAQEREREZFoGECIiIiIiEg0DCBERERERCQaBhAiIiIiIhINAwgREREREYmGAYSIiIiIiETDAEJERERERKJhACEiIiIiItEwgBARERERkWgYQIiIiIiISDQMIEREREREJBoGECIiIiIiEg0DCBERERERiYYBhIiIiIiIRMMAQkREREREomEAISIiIiIi0TCAEBERERGRaBhAiIiIiIhINAwgREREREQkGgYQIiIiIiISDQMIERERERGJhgGEiIiIiIhEwwBCRERERESiYQAhIiIiIiLRMIAQEREREZFoGECIiIiIiEg0DCBERERERCSa/weYjE0rf8rsXQAAAABJRU5ErkJggg=="`;
